spec/support/idv_examples/gpo_otp_verification.rb

```diff
@@ -0,0 +1,96 @@
+shared_examples 'gpo otp verification' do |options|
+  include IdvStepHelper
+
+  let(:expected_redirect_after_gpo_otp_verification) do
+    if options && options[:redirect_after_verification]
+      options[:redirect_after_verification].is_a?(Symbol) ?
+        send(options[:redirect_after_verification]) :
+        options[:redirect_after_verification]
+    end
+  end
+
+  let(:account_should_be_verified) do
+    !options || options[:account_should_be_verified] || options[:account_should_be_verified].nil?
+  end
+
+  let(:profile_should_be_active) do
+    !options || options[:profile_should_be_active] || options[:profile_should_be_active].nil?
+  end
+
+  let(:expected_deactivation_reason) do
+    options && options[:expected_profile_deactivation_reason]
+  end
```

Comment:
First time using `shared_examples` here -- does this make sense as a way of passing in options?

Reply:
Shared examples have access to the scope in which they're added, so you can either define a method or use the `let` values from there.

I _think_ this should work as you'd expect:

```rb
shared_examples 'gpo otp verification' do
  let(:profile_should_be_active) { true }
  # ...
end

# ...

context 'ThreatMetrix says "reject"' do
  let(:profile_should_be_active) { true }

  it_behaves_like 'gpo otp verification'
end
```

Or an example of using methods to assert on the expected outcome which is defined as a context-specific method:

https://github.com/18F/identity-idp/blob/08508e30ce6adb1ceb806a96c9ede712b21f4391/spec/features/phone/confirmation_spec.rb#L7-L57

Reply:
Cool, thanks, I think I got this into 2a9c296b40ea627c64711aa766e8c76705c937a2